src/plugins/Trace.ts

```diff
@@ -188,9 +188,18 @@ export class Trace {
 
         const spanList = this.tracer.getRecorder().getSpanList();
         const sampled = (this.config && this.config.samplerConfig) ? this.config.samplerConfig.isSampled(this.rootSpan) : true;
+        const customSampler = (this.config && this.config.samplerConfig && this.config.samplerConfig.customSampler) ?
+                            this.config.samplerConfig.customSampler : false;
+
         if (sampled) {
             for (const span of spanList) {
                 if (span) {
+                    if (customSampler && !customSampler.isSampled(span)) {
```

Comment:
Individual span should not be sampled but root span. Because it might break parent-child relation between spans and leads to problems.

Reply:
So if root span is not sampled, all of the others spans should not be sampled and ignored

Reply:
User should explicitly set `runCustomSamplerOnEachSpan` to apply custom sampler on each span and the responsibility will be on them. When `runCustomSamplerOnEachSpan` is not set
or set the false, custom sampler is applied to only root span and root span is not sampled no chlid span is sampled.

The configuration will look like this.

```
const config = {
    traceConfig: {
        samplerConfig: {
            runCustomSamplerOnEachSpan: true,
            customSampler:{
                isSampled(span) {
                    // custom sample logic is applied here
                    return true;
                }
            }
        }
    }
}
```